app/Rules/ValidProxyConfigFilename.php

```
@@ -0,0 +1,73 @@
+    /**
+     * Run the validation rule.
+     *
+     * Validates proxy configuration filename:
+     * - Must be 1-255 characters
+     * - No path separators (/, \) to prevent path traversal
+     * - Cannot start with a dot (hidden files)
+     * - Only alphanumeric characters, dashes, underscores, and dots allowed
+     * - Must have a basename before any extension
+     * - Cannot use reserved filenames
+     */
+    public function validate(string $attribute, mixed $value, Closure $fail): void
```

PHPMD (2.15.0)

Avoid unused parameters such as '$attribute'. (undefined)

(UnusedFormalParameter)